### PR TITLE
Convert to TypeScript

### PR DESCRIPTION
**Pull request by lewisjb**
Flow isn't good enough yet.

Also updated the build scripts so that start and build _just work_ with sass

### CODE REVIEW

File: client/README.md [hunk 1]
```diff
@@ -2209,7 +2223,21 @@ GitHub Pages doesn’t support routers that use the HTML5 `pushState` history AP
 * You could switch from using HTML5 history API to routing with hashes. If you use React Router, you can switch to `hashHistory` for this effect, but the URL will be longer and more verbose (for example, `http://user.github.io/todomvc/#/todos/42?_k=yknaj`). [Read more](https://reacttraining.com/react-router/web/api/Router) about different history implementations in React Router.
 * Alternatively, you can use a trick to teach GitHub Pages to handle 404 by redirecting to your `index.html` page with a special redirect parameter. You would need to add a `404.html` file with the redirection code to the `build` folder before deploying your project, and you’ll need to add code handling the redirect parameter to `index.html`. You can find a detailed explanation of this technique [in this guide](https://github.com/rafrex/spa-github-pages).
 
+<<<<<<< HEAD
```
Comment (me4502) [inline client/README.md]: 🤔

**Review by me4502 — CHANGES_REQUESTED** [review]

File: client/README.md (hunk 1 above)
Comment (lewisjb) [reply client/README.md]: I... I don't even.. what?

**Comment by byronmejia** [comment]
Why not just use angular? ;)

**Comment by mitchazj** [comment]
Lol

**Pull request merged**